src/admin/index.js

```diff
@@ -620,6 +619,88 @@ module.exports = ({
     return { groups }
   }
 
+  /**
+   * Delete groups in a broker
+   *
+   * @param {string[]} [groupIds]
+   * @return {Promise<DeleteGroups>}
+   *
+   * @typedef {Array} DeleteGroups
+   * @property {string} groupId
+   * @property {number} errorCode
+   */
+  const deleteGroups = async groupIds => {
+    if (!groupIds || !Array.isArray(groupIds)) {
+      throw new KafkaJSNonRetriableError(`Invalid groupIds array ${groupIds}`)
+    }
+
+    const invalidGroupId = groupIds.some(g => typeof g !== 'string')
+
+    if (invalidGroupId) {
+      throw new KafkaJSNonRetriableError(`Invalid groupId name: ${JSON.stringify(invalidGroupId)}`)
+    }
+
+    const retrier = createRetry(retry)
+
+    const results = []
+
+    const clonedGroupIds = groupIds.slice()
+
+    return retrier(async (bail, retryCount, retryTime) => {
+      try {
+        if (clonedGroupIds.length === 0) return []
+
+        await cluster.refreshMetadata()
+
+        const brokersPerGroups = {}
+        const brokers = {}
+        for (const groupId of clonedGroupIds) {
+          const broker = await cluster.findGroupCoordinator({ groupId })
+          if (brokersPerGroups[broker.nodeId] === undefined) brokersPerGroups[broker.nodeId] = []
+          brokersPerGroups[broker.nodeId].push(groupId)
+          brokers[broker.nodeId] = broker
+        }
+
+        for (const nodeId in brokersPerGroups) {
+          try {
+            const res = await brokers[nodeId].deleteGroups(brokersPerGroups[nodeId])
```

Comment:
Now this is not done in parallel anymore. To get parallelism back, you can just map over the promises and catch any potential error (basically `Promise.allSettled`, but with Node 10 support. Essentially:

```javascript
const allSettled = promises => Promise.all(promises.map(promise => promise
    .then(value => ({ status: 'fulfilled', value }))
    .catch(reason => ({ status: 'rejected', reason })))
)

const results = allSettled(brokersPerGroups.map(nodeId => brokers[nodeId].deleteGroups(brokersPerGroups[nodeId])))
const { resolved, rejected } = results.reduce((promises, promise) => {
  if (promise.status === 'fulfilled') {
    promises.resolved.push(promise)
  } else {
    promises.rejected.push(promise)
  }
  return promises
}, { resolved: [], rejected: []})
```

Reply:
I did a new implementation, I think this one is better one as this gives all unsuccessful deletions in array but does not end if one deletion fails, so you are sure that if groupId is not in error it is deleted, sorry for late implementation

Reply:
Looks much better